### PR TITLE
1.21 codebase rebrand

### PR DESCRIPTION
**Pull request by barrystyle**
Collective changes migrating Bitcoin 0.21 branch for use with Dogecoin.
Includes filelevel, cli and tests.

### CODE REVIEW

**Comment by rnicoll** [comment]
There's a bunch of internal references that need to be updated to match. This won't be all of them, but as a starting point:

```
~/dogecoin$ grep -r bitcoind src .gitignore | grep \"
src/bitcoin-cli.cpp:    argsman.AddArg("-rpcwallet=<walletname>", "Send RPC for non-default wallet on RPC server (needs to exactly match corresponding -wallet option passed to bitcoind). This changes the RPC endpoint used, e.g. http://127.0.0.1:8332/wallet/<walletname>", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
src/bitcoin-cli.cpp:            throw std::runtime_error("-netinfo requires bitcoind server to be running v0.21.0 and up");
src/bitcoin-cli.cpp:        throw CConnectionFailed(strprintf("Could not connect to the server %s:%d%s\n\nMake sure the bitcoind server is running and that you are connecting to the correct RPC port.", host, port, responseErrorMessage));
src/bitcoind-res.rc:            VALUE "FileDescription",    "bitcoind (Bitcoin node with a JSON-RPC server)"
src/bitcoind-res.rc:            VALUE "InternalName",       "bitcoind"
src/bitcoind-res.rc:            VALUE "OriginalFilename",   "bitcoind.exe"
src/bitcoind-res.rc:            VALUE "ProductName",        "bitcoind"
src/init.cpp:static const char* BITCOIN_PID_FILENAME = "bitcoind.pid";
src/wallet/test/wallet_tests.cpp:                      "by pruning or data corruption (see bitcoind log for details) and could be dealt with by "
src/wallet/sqlite.cpp:        throw std::runtime_error("SQLiteDatabase: Unable to obtain an exclusive lock on the database, is it being used by another bitcoind?\n");
src/wallet/rpcwallet.cpp:                "\nNote that all wallet command-line options used when starting bitcoind will be"
src/wallet/rpcwallet.cpp:            throw JSONRPCError(RPC_METHOD_DEPRECATED, "Using bumpfee with wallets that have private keys disabled is deprecated. Use psbtbumpfee instead or restart bitcoind with -deprecatedrpc=bumpfee. This functionality will be removed in 0.22");
src/wallet/rpcdump.cpp:                                      "caused by pruning or data corruption (see bitcoind log for details) and could "
src/wallet/rpcdump.cpp:                                      "caused by pruning or data corruption (see bitcoind log for details) and could "
src/bitcoind.cpp:            strUsage += "\nUsage:  bitcoind [options]                     Start " PACKAGE_NAME "\n";
src/bitcoind.cpp:                return InitError(Untranslated(strprintf("Command line contains unexpected token '%s', see bitcoind -h for a list of options.\n", argv[i])));
```

**Comment by barrystyle** [comment]
Sure, understood. Just at the moment before we merge most of the functionality in, i'd like to keep the commits as 'clean' as we can get them - before a thorough search and replace renders the commits unusable for the future.

**Comment by barrystyle** [comment]
Could i get this approved?

**Comment by barrystyle** [comment]
Yeah can I get this merged?

**Comment by rnicoll** [comment]
Main issue is we need to confirm Travis still runs, and I'm not sure why it hasn't. Going to close then re-open to try forcing it.

**Comment by nolim1t** [comment]
much changes wow

**Comment by rnicoll** [comment]
Okay looks like you need to enable Travis on _your_ repo, for us to get the review. I've fixed some of the CI issues and pushed that, meanwhile.

**Comment by barrystyle** [comment]
image
travis just seems to stop after the tests, have corrected them.. see if it works now

**Comment by barrystyle** [comment]
good to go

**Comment by langerhans** [comment]
Well now all gui related tests are disabled. At least on CI. What's the plan for them?

**Comment by barrystyle** [comment]
all tests run, but as i've said travis just stops after all tests complete.
you're free to git clone and try them yourself if you like.

**Comment by barrystyle** [comment]
Yep was just checking this; these are on master but not backported to 0.21:
https://github.com/bitcoin/bitcoin/commit/86968882a8a26312a7af29c572313c4aff488c11
https://github.com/bitcoin/bitcoin/commit/388053e1722632c2e485c56a444bc75cf0152188
https://github.com/bitcoin/bitcoin/commit/173cc9b7be335b5dd2cc1bb112dfa6ec5c13ec12

**Comment by barrystyle** [comment]
I've got it working with cherry-picked #fae32f295cc5b57c1cb95090bb60cddb42f9778a, mind if i force push?

**Comment by rnicoll** [comment]
> 
> 
> I've got it working with cherry-picked [bitcoin/bitcoin@fae32f2](https://github.com/bitcoin/bitcoin/commit/fae32f295cc5b57c1cb95090bb60cddb42f9778a), mind if i force push?

Go for it, I can always rebase some of my stuff that fixes up bitcoind to dogecoind in the tests

**Comment by barrystyle** [comment]
> 
> 
> > I've got it working with cherry-picked [bitcoin/bitcoin@fae32f2](https://github.com/bitcoin/bitcoin/commit/fae32f295cc5b57c1cb95090bb60cddb42f9778a), mind if i force push?
> 
> Go for it, I can always rebase some of my stuff that fixes up bitcoind to dogecoind in the tests

I pulled your changes down first, so all good :)

**Comment by barrystyle** [comment]
Requires https://github.com/bitcoin/bitcoin/pull/19137 by the looks.. merging

**Comment by barrystyle** [comment]
All checks passing.

**Review by langerhans — APPROVED** [review]
Changes looks good, thanks for fixing the tests! Going forward I do wonder if we really need to change comments, but I figure these happened automatically.

File: configure.ac [hunk 1]
```diff
@@ -8,7 +8,7 @@ define(_CLIENT_VERSION_IS_RELEASE, true)
 define(_COPYRIGHT_YEAR, 2020)
 define(_COPYRIGHT_HOLDERS,[The %s developers])
 define(_COPYRIGHT_HOLDERS_SUBSTITUTION,[[Bitcoin Core]])
```
Comment (dan-ryan) [inline configure.ac:10]: Use `[Dogecoin Core]`?

File: src/Makefile.am [hunk 2]
```diff
@@ -572,16 +572,16 @@ nodist_libbitcoin_util_a_SOURCES = $(srcdir)/obj/build.h
 #
 
 # bitcoind & bitcoin-node binaries #
```
Comment (dan-ryan) [inline src/Makefile.am:574]: Update comment?

File: configure.ac (hunk 1 above)
Comment (barrystyle) [reply configure.ac:10]: Not replacing copyrights, sorry.

File: src/Makefile.am (hunk 2 above)
Comment (barrystyle) [reply src/Makefile.am:574]: The changes present in Makefile.am are purely functional; there is no point changing something that does not affect the build directly, that could create a future merge issue, no matter how insignificant.

**Comment by barrystyle** [comment]
So.. ready to merge?

**Comment by rnicoll** [comment]
Squashed in, thanks!

**Pull request merged**